pkg/scheduler/objects/node.go

```diff
@@ -353,6 +356,8 @@ func (sn *Node) addAllocationInternal(alloc *Allocation, force bool) bool {
 		sn.allocations[alloc.GetAllocationKey()] = alloc
 		sn.allocatedResource.AddTo(res)
 		sn.availableResource.SubFrom(res)
+		sn.allocatedResource.Prune()
+		sn.availableResource.Prune()
```

Comment:
The alloc should not have zeros in it. The only prune that is needed is when we do the `SubFrom`

Reply:
Addressed in latest PR, thanks!